### ISSUE TITLE
CraftQL support

### ISSUE THREAD

**Issue by pr1ntr** [spicywebau/craft-neo#173]
I was testing this for my project. It turns out that it does not add a schema for CraftQL which is problematic for us.

**Comment by markhuot** [comment]
I have a basic implementation of this working over at,

https://github.com/spicywebau/craft-neo/compare/master...markhuot:craftql?expand=0

@ttempleton, would you entertain merging this in to Neo? Or would you rather I keep this outside of Neo, in CraftQL itself?

**Comment by ttempleton** [comment]
Thanks @markhuot!  Yeah, we're happy to merge it in to Neo.

**Comment by seamofreality** [comment]
This would be absolutely great!
Looking forward to the merge.

**Comment by wiledal** [comment]
Any indication for when the merge might happen?
We are working on a site that utilizes CraftQL, and it would be a tremendous improvement if we could use Neo with it.

**Comment by remmar13** [comment]
> Any indication for when the merge might happen?
> We are working on a site that utilizes CraftQL, and it would be a tremendous improvement if we could use Neo with it.

Same here! 👍

**Comment by wiledal** [comment]
As it's been over 3 months with no update, I did an unconventional composer install from a fork of @markhuot's branch, to see if I could potentially cobble together the support until it gets officially merged (we really want this functionality), but I get this GraphQL error as soon as there is a `Neo Group` field present in a section:

```json
{
  "error": "Return value of markhuot\\CraftQL\\Builders\\Field::getTypeConfig() must be an instance of GraphQL\\Type\\Definition\\Type, instance of markhuot\\CraftQL\\Builders\\Union returned"
}
```

You could potentially disregard this issue as invalid, looking at the hacky install method, but one could still assume the fork would work?

**Comment by nstCactus** [comment]
I managed to get basic Neo-field support using the [following module](https://gist.github.com/nstCactus/199e339b56bc27befff369f771f1f53a).
Currently this fits my needs so I didn't dig any deeper but it could be used as a starting point.

**Comment by tomfrew** [comment]
Would also love to see this get merged!

I've tried manually applying @markhuot's changes from https://github.com/spicywebau/craft-neo/compare/master...markhuot:craftql to the latest version and seems to work great. 
@wiledal I didn't run into your issue, although haven't tried installing from a folk only manually updating the plugin.

**Comment by tomfrew** [comment]
Although I can't get dropdown or multi-select fields to show in the schema

**Comment by puck3000** [comment]
@tomfrew: Thank you for your solution, seems to work so far!!!

**Comment by lastgiven** [comment]
@markhuot Is there any workaround at the moment to get this working? 

Is third party field support a viable solution for the time being?

**Comment by mediabeastnz** [comment]
+1 now that 3.3 is out

**Comment by Jones-S** [comment]
Is this already available?
I really need NEO fields in GraphQL :).

**Comment by Jones-S** [comment]
I really like @tomfrew s approach but it was some time back and I wanted to make sure I have the newest neo version and on top the changes from @markhuot.
So I made another fork, added the PR manually (I did not know how to merge an already merged PR... couldn't pull it) and published here: 
https://github.com/wearelucid/craft-neo/

My composer looks like this:
```
...
"require": {
    "spicyweb/craft-neo": "dev-master",
  },
"repositories": [
    {
      "type": "vcs",
      "url": "git@github.com:wearelucid/craft-neo.git"
    }
  ]
```
Note also that I am still on craft 3.2.8, because 3.3 includes graphql in core. But there were too many plugins not supported within graphql, so I stuck with 3.2.8 with craftql for now.
🤷‍♂️

**Comment by Jones-S** [comment]
Do you think when using the core graphql this is resolved? (Meaning with the newest version: 2.5.3) (I can't update yet, so I can't test the __typename thing, because of some other dependencies not supporting the core graphql.. but maybe someone knows)...

**Comment by ttempleton** [comment]
Craft Pro has had in-built GraphQL support for a while now (which Neo supports), and CraftQL hasn't been updated since last August, so I'm going to close this issue now and announce that we no longer have plans to include CraftQL support in Neo.  I understand that this might be disappointing for any Neo users who are still using CraftQL with Craft Solo, but we feel it would not be responsible to introduce compatibility with another plugin at a time when that plugin appears to no longer be maintained.  If CraftQL is updated in the future, we will reconsider this decision.
